File: 242-valid-anagram/valid-anagram.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        if len(s) == 0 and len(t) == 0:
            return True

        if len(t) != len(s):
            return False

        s_word_count = defaultdict(int)
        t_word_count = defaultdict(int)

        for i in range(len(s)):
            s_word_count[s[i]] += 1
            t_word_count[t[i]] += 1
        
        return s_word_count == t_word_count

File: 242-valid-anagram/valid-anagram.java
class Solution {
    public boolean isAnagram(String s, String t) {
        if (s.length() != t.length()) {
            return false;
        }

        Map<Character, Integer> s_count = new HashMap<>();
        Map<Character, Integer> t_count = new HashMap<>();

        for (int i = 0; i < s.length(); i++) {
            int tmp_s_count = s_count.getOrDefault(s.charAt(i), 0);
            s_count.put(s.charAt(i), tmp_s_count+1);
            int tmp_t_count = t_count.getOrDefault(t.charAt(i), 0);
            t_count.put(t.charAt(i), tmp_t_count+1);
        }

        return s_count.equals(t_count);
    }
}